flowmachine/flowmachine/core/server/action_handlers.py

```diff
@@ -277,6 +277,41 @@ def action_handler__get_geography(aggregation_unit):
     return ZMQReply(status="success", payload=payload)
 
 
+def action_handler__get_available_dates(event_types=None):
+    """
+    Handler for the 'get_available_dates' action.
+
+    Returns a dict of the form {"calls": [...], "sms": [...], ...}.
+
+    Parameters
+    ----------
+    event_types: list of str, optional
+        List of event types for which to return available dates.
+
+    Returns
+    -------
+    ZMQReply
+        The reply from the action handler.
+    """
+    conn = Query.connection
+    if event_types is None:
+        event_types = tuple(
+            sorted([table_name for table_name, _, _, _ in conn.available_tables])
+        )
+    elif isinstance(event_types, (list, tuple)):
+        event_types = tuple(event_types)
+    else:
+        return ZMQReply(
+            status="error", msg=f"Invalid value for argument `event_types`."
+        )
+
+    available_dates = {
+        event_type: [date.strftime("%Y-%m-%d") for date in dates]
+        for (event_type, dates) in conn.available_dates(table=event_types).items()
```

Comment:
Did you come down on whether to use `strictness==2` here? Seems to me no reason to ever say that there is an available date if no data!? Could cause a user to ask for some other query type for that date and then get nothing back which would be confusing.

Reply:
Good point, I'm happy to use `strictness=2`. Any objections @greenape?